lib/datasource/npm/get.ts

```diff
@@ -93,6 +94,27 @@ export async function getDependency(
     if (res.repository?.directory) {
       dep.sourceDirectory = res.repository.directory;
     }
+
+    // Massage the repository URL for non-compliant strings (see issue #4610)
+    // Remove the non-compliant segments of path, so the URL looks like "<scheme>://<domain>/<vendor>/<repo>"
+    // and add directory to the repository
+    const sourceUrlCopy = `${sourceUrl}`;
+    const sourceUrlSplit: string[] = sourceUrlCopy.split('/');
+
+    if (sourceUrlSplit.length > 7) {
```

Comment:
I don't think we can be sure that this is a GitHub repo at this point. This massage should only be for github for now so it needs an extra check

Reply:
Hi @rarkins that's a good idea. I've added the check:

```
 if (sourceUrlSplit.length > 7 && sourceUrlSplit[2] === 'github.com') {
      /* istanbul ignore next */
      if (dep.sourceDirectory) {
        logger.debug(
          { dependency: packageName },
          `Ambiguity: dependency has the repository URL path and repository/directory set at once; have to override repository/directory`
        );
      }
      dep.sourceUrl = sourceUrlSplit.slice(0, 5).join('/');
      dep.sourceDirectory = sourceUrlSplit
        .slice(7, sourceUrlSplit.length)
        .join('/');
    }
```

Also I added a test for checking that only github repos are massaged.